client/src/client.rs

```diff
@@ -329,7 +333,64 @@ pub trait RpcApi: Sized {
     /// Returns a data structure containing various state info regarding
     /// blockchain processing.
     fn get_blockchain_info(&self) -> Result<json::GetBlockchainInfoResult> {
-        self.call("getblockchaininfo", &[])
+        let mut raw: serde_json::Value = self.call("getblockchaininfo", &[])?;
+        // The softfork fields are not backwards compatible:
+        // - 0.18.x returns a "softforks" array and a "bip9_softforks" map.
+        // - 0.19.x returns a "softforks" map.
+        Ok(if self.version()? < 190000 {
```

Comment:
Would caching version be a good idea? Otherwise this call would always be two actual calls. Or maybe just don't check the version and first attempt deserialization to newer format, and if fail, try the previous one? Or something like that? This way the newer  version takes no perf hit.

Reply:
Good point. I thought about caching the version. But then we'd have an extra requirement that the underlying daemon can't change for the existence of the client. That sounds reasonable, but some people might want to relax that.

Reply:
@stevenroose BTW. You already have that requirement in a way. Just less chance of triggering it.

Reply:
I created this: https://github.com/rust-bitcoin/rust-bitcoincore-rpc/issues/99